### PR TITLE
Adding for await of statement example

### PR DESCRIPTION
**Pull request by wmsbill**
This PR aims to add a new example for `for await...of` as a supporting example for the `for await...of` statement MDN docs page.

### CODE REVIEW

**Comment by schalkneethling** [comment]
@wmsbill Because of the following security vulnerability https://snyk.io/blog/malicious-code-found-in-npm-package-event-stream that was revealed yesterday, we are reaching out to everyone that currently has an open PR on interactive examples.

We ask that you please update your fork's master branch, and rebase your current feature branch against the new master. This will update the affected dependency to a version that is no longer affected, and remove the security concern.

We appreciate your assistance in this matter. Should you need any assistance, please feel free to reach out.

File: live-examples/js-examples/statement/statement-forawaitof.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<pre>
+<code id="static-js">async function* foo(){
+  yield 1;
+  yield 2;
+}
+
+(async function () {
+  for await (let num of foo()) {
```
Comment (Elchi3) [inline live-examples/js-examples/statement/statement-forawaitof.html:8]:
```suggestion
  for await (const num of foo()) {
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]
I'm very sorry this took forever. I'm finally cleaning my review queue.
I guess this could still land. `const` seems generally preferable, though.

**Review by Elchi3 — APPROVED** [review]
Thank you! 👍

**Pull request merged**